TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/tests/test_source_cluster_multiversion.py

```diff
@@ -0,0 +1,139 @@
+        cluster.call_api(f"/{TEST_INDEX_NAME}/_doc/{i + 1}", HttpMethod.PUT,
+                         data=TEST_DOC, headers={"Content-Type": "application/json"})
+
+
+def seed_data_with_types(cluster: Cluster, doc_count):
+    # Create index with settings
+    cluster.call_api(f"/{TEST_INDEX_NAME}", HttpMethod.PUT,
+                     data=json.dumps({"settings": {"index": {"number_of_shards": 1, "number_of_replicas": 0}}}),
+                     headers={"Content-Type": "application/json"})
+    
+    doc_type = "document"
+    for i, _ in enumerate(range(doc_count)):
+        cluster.call_api(f"/{TEST_INDEX_NAME}/{doc_type}/{i + 1}", HttpMethod.PUT,
+                         data=TEST_DOC, headers={"Content-Type": "application/json"})
+
+
+def setup_snapshot_repo_and_snapshot(cluster: Cluster):
+    # Register repo
+    cluster.call_api("/_snapshot/migration_assistant_repo", HttpMethod.PUT,
+                     data=json.dumps({"type": "fs", "settings": {"location": CLUSTER_SNAPSHOT_DIR}}),
+                     headers={"Content-Type": "application/json"})
+    cluster.call_api(f"/_snapshot/migration_assistant_repo/{SNAPSHOT_NAME}", HttpMethod.POST)
+
+
+@pytest.fixture()
+def env_with_source_container(request):
+    version = request.param
+    # Spin up the Elasticsearch container and wait until it's healthy
+    container = SearchContainer(version, mem_limit="3G")
+    container.start()
+
+    base_url = f"http://{container.get_container_host_ip()}:{container.get_exposed_port(9200)}"
+    # Create a temporary services.yaml file based on the services.yaml spec
+    services_config = {
+        'source_cluster': {
+            'endpoint': base_url,
+            'allow_insecure': True,
+            'no_auth': {}
+        },
+        'target_cluster': {
+            'endpoint': "http://this_endpoint_does_not_exist:9200",
+            'allow_insecure': True,
+            'no_auth': {}
+        },
+        'snapshot': {
+            'snapshot_name': SNAPSHOT_NAME,
+            'fs': {
+                'repo_path': CLUSTER_SNAPSHOT_DIR
+            }
+        }
+    }
+
+    with tempfile.NamedTemporaryFile(mode='w', delete=False) as temp_config:
+        yaml.dump(services_config, temp_config)
+        temp_config_path = temp_config.name
+
+    logging.basicConfig(level=logging.INFO)
+
+    env = Environment(temp_config_path)
+    if version.flavor == "ELASTICSEARCH" and version.major_version == 5:
+        seed_data_with_types(env.source_cluster, DOC_COUNT)
+    else:
+        seed_data_without_types(env.source_cluster, DOC_COUNT)
+
+    setup_snapshot_repo_and_snapshot(env.source_cluster)
+
+    yield env
+    # Stop the container and clean up the temporary services.yaml file after tests complete
+    container.stop()
+    os.remove(temp_config_path)
+
+
+@pytest.mark.parametrize("env_with_source_container,refresh,json",
+                         itertools.product(SUPPORTED_SOURCE_CLUSTERS, [True, False], [True, False]),
```

Comment:
Rather that have a flag to refresh or not, can we minimize the parameterized values so that we reuse the same cluster to do that true case and the false case?  This will lower the runtime considerably for each of these pairs that we remove.

Reply:
Oh, yeah, basically run with refresh first, and then non-refresh in the same test case? 👍